src/panels/lovelace/components/hui-image.ts

```diff
@@ -159,12 +161,25 @@ export class HuiImage extends LitElement {
   }
 
   protected updated(changedProps: PropertyValues): void {
+    this._checkConnection();
+
     if (changedProps.has("cameraImage") && this.cameraView !== "live") {
       this._updateCameraImageSrc();
       this._startUpdateCameraInterval();
     }
   }
 
+  private async _checkConnection(): Promise<void> {
+    if (this.hass!.connected) {
+      await this._updateCameraImageSrc();
+      this._startUpdateCameraInterval();
+    } else {
+      this._stopUpdateCameraInterval();
+      this._cameraImageSrc = undefined;
+    }
+    this._hassConnected = this.hass!.connected;
```

Comment:
```suggestion
```

Reply:
There's a specific reason for using another prop. Because the render happens as soon as the connected bool is set, the previous token is still used/cached.

![image](https://user-images.githubusercontent.com/28114703/91637642-d48b7e80-ea01-11ea-884f-0d273a7c876b.png)


Using this extra property makes sure the image has updated first, before the render is allowed and auth fails

Reply:
There might be a smarter way of doing this without an extra prop. I'll take a look

Reply:
You are updating `_cameraImageSrc` right? That should trigger a new update.